src/host/kvpair.rs

```diff
@@ -202,6 +202,53 @@ impl<const DEPTH: usize> MongoMerkle<DEPTH> {
         }
         Ok(())
     }
+
+    fn generate_default_node(
+        &self,
+        index: u32,
+        hash: &[u8; 32],
+    ) -> Result<MerkleRecord, MerkleError> {
+        let height = (index + 1).ilog2();
+        let default = self.get_default_hash(height as usize)?;
+        let child_hash = if height == Self::height() as u32 {
+            [0; 32]
+        } else {
+            self.get_default_hash((height + 1) as usize)?
+        };
+        if default == *hash {
+            Ok(MerkleRecord {
+                index,
+                hash: default,
+                data: [0; 32],
+                left: child_hash,
+                right: child_hash,
+            })
+        } else {
+            Err(MerkleError::new(*hash, index, MerkleErrorCode::InvalidHash))
+        }
+    }
+
+    fn check_generate_default_node(
+        &self,
+        index: u32,
+        hash: &[u8; 32],
+    ) -> Result<(MerkleRecord, bool), MerkleError> {
+        let mut exist = false;
+        let node = self
+            .get_record(index, hash)
+            .expect("Unexpected DB Error")
+            .map_or_else(
+                || -> Result<MerkleRecord, MerkleError> {
+                    Ok(self.generate_default_node(index, hash)?)
+                },
+                |x| -> Result<MerkleRecord, MerkleError> {
+                    exist = true;
+                    assert_eq!(x.index, index);
```

Comment:
assert_eq! should not be used in non-test part I think.
For this one, what I suggested the name of check_generate_default_node() is because we do hash check in the function.
In fact for generate_default_node, we only need index as the input, we do not need hash as input. I prefer have the function name which can exactly represent the job it is doing.

Thus in fact, we can remove generate_default_node() function and just add check_generate_default_node() ( implement can be just like the first pr codes）, which do the "if default == *hash"  check (no need do x.index == index check as it is set by this function）

Or if you think in the future maybe 
fn generate_default_node(
        &self,
        index: u32,
    )
 is a possible reusable function, you can add it but for this one, no need input hash as we just generate a default empty node based on index.